addons/mail/static/src/core/common/composer.js

```diff
@@ -732,10 +735,53 @@ export class Composer extends Component {
 
     saveContent() {
         const composer = toRaw(this.props.composer);
-        const fullComposerContent =
-            document
-                .querySelector(".o_mail_composer_form_view .note-editable")
-                ?.innerText.replace(/(\t|\n)+/g, "\n") ?? composer.text;
+        const editable = document.querySelector(".o_mail_composer_form_view .note-editable");
+        let fullComposerContent;
+        if (editable) {
+            /**
+             * @param {HTMLElement} element
+             * @returns {string}
+             */
+            const toText = element => {
+                let text = "";
+                for (const child of element.childNodes) {
+                    if (child.nodeType === Node.TEXT_NODE) {
+                        text += child.textContent;
+                    } else if (child.nodeType === Node.ELEMENT_NODE) {
+                        if (child.classList.contains("email-signature")) {
+                            continue; // Ignore the email signature
+                        }
+                        if (child.nodeName === "UL") {
+                            if (child.classList.contains("o_checklist")) {
+                                for (const li of child.childNodes) {
+                                    if (li.classList.contains("o_checked")) {
+                                        text += `\t[x] ${li.textContent}\n`;
+                                    } else {
+                                        text += `\t[ ] ${li.textContent}\n`;
+                                    }
+                                }
+                            } else {
+                                for (const li of child.childNodes) {
+                                    text += `\t- ${li.textContent}\n`;
+                                }
+                            }
+                        } else if (child.nodeName === "OL") {
+                            let index = 1;
+                            for (const li of child.childNodes) {
+                                text += `\t${index}. ${li.textContent}\n`;
+                                index++;
+                            }
+                        } else {
+                            text += `${toText(child)}\n`;
+                        }
+                    }
+                }
+                return text;
+            };
+            fullComposerContent = toText(editable).trimEnd();
```

Comment:
Note: When I clone the editable zone (using `cloneNode(true)` https://developer.mozilla.org/en-US/docs/Web/API/Node/cloneNode), remove the email signature block from the cloned node and use `innerText` or `textContent` to retrieve the text, the line breaks get discarded and I don't know how I can preserve them.

The only solution I find is to write a custom function to generate text from the editable zone: The function recursively traverses the dom hierarchy and generates text as we want. **Do you think it would be okay to do that?**